bom/build.gradle.kts

```
@@ -53,7 +53,7 @@ dependencies {
 val defaultScalaVersion: String = project.findProperty("defaultScalaVersion")!!.toString()
 val scalaVersions: List<String>? = project.findProperty("supportedScalaVersions")?.toString()?.split(",")
 
-assert(!scalaVersions.isNullOrEmpty()) {
+require(!scalaVersions.isNullOrEmpty()) {
     "Scala versions must be provided as a comma-separated list in the 'supportedScalaVersions' project property"
 }
 
```

I suggest adding a check for the expected version format - this way, if the version is accidentally changed during development (e.g., an extra patch version is added), it can be caught here in the BOM.

```suggestion
scalaVersions?.forEach { version ->
    require(version.matches(Regex("\\d\\.\\d{2}"))) {
        "Scala version '$version' must be in the format X.YY"
    }
}
```